### PR TITLE
Save and restore format dialog title rather than duplicating it

### PR DESCRIPTION
**Pull request by malxau**
This is to a fix things noted in #149 .

In the current code, the format select dialog is shown, which invokes `SHFormatDrive`, and that function can change the title of the parent window, so the code restores the title from `IDS_FORMATSELECTDLGTITLE`.  In other words, the dialog's title is recorded twice, once in the dialog resource (used when it is displayed) and once in `IDS_FORMATSELECTDLGTITLE` which is used on the dialog if `SHFormatDrive` fails to complete successfully.

Unsurprisingly, these two strings became out-of-sync as a result of this duplication, despite being intended to be the same string.

This PR does three things:

- Save and restore the title across the call to `SHFormatDrive`, which eliminates the need for `IDS_FORMATSELECTDLGTITLE` so titles are stored once.
- Use a memory allocation rather than `CCH_DLG_TITLE` so it will function regardless of the length of the dialog title.  #149 comments how `CCH_DLG_TITLE` should be reviewed to verify it's large enough for all languages, and it wasn't large enough for de-DE, so on return from `SHFormatDrive` the dialog title was being truncated.
- Use the ja-JP version of `IDS_FORMATSELECTDLGTITLE` as the dialog's title, which had not been localized.  For other languages it seemed preferable to keep the dialog title since that's the obvious one that everybody sees so is more likely to be correct; for ja-JP it clearly wasn't correct.  I'd still appreciate verification from @mattn that this string is correct though.

### CODE REVIEW

**Comment by craigwims** [comment]
@malxau-msft, if you can fix the conflict, one of us can merge it.

File: src/wfdlgs3.c [hunk 1]
```diff
@@ -788,6 +788,18 @@ FormatSelectDlgProc(register HWND hDlg, UINT wMsg, WPARAM wParam, LPARAM lParam)
                 // that parent, even if it is hidden.
                 ShowWindow(hDlg, SW_HIDE);
 
+                pszDlgTitle = NULL;
+                dwCchTitleLength = GetWindowTextLength(hDlg);
+                if (dwCchTitleLength > 0)
+                {
+                    dwCchTitleLength++;
+                    pszDlgTitle = malloc(ByteCountOf(dwCchTitleLength));
+                    if (pszDlgTitle != NULL)
```
Comment (craigwims) [inline src/wfdlgs3.c:797]: sorry; just noticed.  The WinFile standard for memory allocation is LocalAlloc().

**Comment by schinagl** [comment]
@malxau: It seems we are just a LocalAlloc() away from Merging....

**Comment by schinagl** [comment]
Even this one could go on ...

**Pull request merged**